contracts/core/SedaCoreV1.sol

```diff
@@ -80,22 +115,101 @@ contract SedaCoreV1 is ISedaCore, RequestHandlerBase, ResultHandlerBase, UUPSUpg
+        // Call parent postResult but return the batch sender for fee distribution
+        (bytes32 resultId, address batchSender) = super.postResultAndGetBatchSender(result, batchHeight, proof);
 
+        // Clean up state
         _removeRequest(result.drId);
-        delete _storageV1().requestTimestamps[result.drId];
+        delete _storageV1().requestDetails[result.drId];
+
+        // Handle fee distribution
+        uint256 refundAmount;
+
+        if (requestDetails.requestFee > 0) {
+            address payableAddress = result.paybackAddress.length == 20
+                ? address(bytes20(result.paybackAddress))
+                : address(0);
+
+            if (payableAddress == address(0)) {
+                // If no payback address, send all request fee to requestor
+                refundAmount += requestDetails.requestFee;
+            } else {
+                // Calculate request fee split and send to payback address
+                uint256 submitterFee = (result.gasUsed * requestDetails.requestFee) / requestDetails.gasLimit;
+                if (submitterFee > 0) {
+                    _transferFee(payableAddress, submitterFee);
+                    emit FeeDistributed(result.drId, payableAddress, submitterFee, ISedaCore.FeeType.REQUEST);
+                }
+                // Update requestor amount with refund (executed at the end of the function)
+                refundAmount += requestDetails.requestFee - submitterFee;
+            }
+        }
+
+        // Send result fee to msg.sender
+        if (requestDetails.resultFee > 0) {
+            _transferFee(msg.sender, requestDetails.resultFee);
+            emit FeeDistributed(result.drId, msg.sender, requestDetails.resultFee, ISedaCore.FeeType.RESULT);
+        }
+
+        // Add batch fee to requestor amount if no valid batch sender
+        if (requestDetails.batchFee > 0) {
+            if (batchSender == address(0)) {
+                // If no batch sender, send all batch fee to requestor
+                refundAmount += requestDetails.batchFee;
+            } else {
+                // Send batch fee to batch sender
+                _transferFee(batchSender, requestDetails.batchFee);
+                emit FeeDistributed(result.drId, batchSender, requestDetails.batchFee, ISedaCore.FeeType.BATCH);
+            }
+        }
+
+        // Send combined amount to requestor if any (refunds)
+        if (refundAmount > 0) {
+            _transferFee(requestDetails.requestor, refundAmount);
+            emit FeeDistributed(result.drId, requestDetails.requestor, refundAmount, ISedaCore.FeeType.REFUND);
+        }
 
         return resultId;
     }
 
+    /// @inheritdoc ISedaCore
+    function increaseFees(
+        bytes32 requestId,
+        uint256 additionalRequestFee,
+        uint256 additionalResultFee,
+        uint256 additionalBatchFee
+    ) public payable override {
+        // Check that amount is equal to sum of fees
```

Comment:
One more example and I'll stop yapping about it

`// We have to ensure the total fees match the expected amount to prevent incorrect payments or underfunding.`